src/docdb/nodes.ts

```diff
@@ -67,13 +71,17 @@ export class DocDBDatabaseNode implements INode {
 
 }
 
-
 export class DocDBCollectionNode implements INode {
 
 	constructor(readonly id: string, readonly db: DocDBDatabaseNode) {
 	}
```

Comment:
Just create the loadmore once in the constructor. Also you only need to pass in the id, not the whole node

```
this._loadMoreNode = new LoadMoreNode(id)
```

Reply:
The ID in itself may not be enough. I use a parentNode reference in the LoadMoreNode constructor. I'll still create it just once, just after constructor.